### ISSUE TITLE
German keyboard adds a ^-sign

### ISSUE THREAD

**Issue by MikeFrom** [jitspoe/godot-console#13]
Hi jitspoe,

nice console. On my German keyboard the ~-key is a ^-key. When I open the console, in the console line a ^-sign is added, this has to be removed always manually.

<img width="63" alt="German_keyboard_trouble" src="https://github.com/user-attachments/assets/f94eb094-631f-4777-9f7f-56b229364c81">

Is there a fix?

Thanks

Mike

**Comment by jitspoe** [comment]
Thanks for the report.  I'll look into it!

**Comment by jitspoe** [comment]
Hmm, I'm not getting that behavior.  For me, I have to type ^ twice, even when typing normally.  I press it once, and nothing happens.  I press it again, and two ^^'s show up (in a normal text box).  In-game, I just have to tap it twice to open or close the console, but I don't get a ^ character showing up.  There's something special about the way the OS handles that key.  What OS are you on, and what exact keyboard language are you using?  If the OS is reporting 2 key presses to Godot, I'm not sure how best to deal with that. 🤔

**Comment by jitspoe** [comment]
Right.  The problem is the Godot engine itself doesn't send the pressed event until the key is pressed a second time.  I don't think I can fix that without modifying the engine.  There's an issue about it on the main Godot repo here: https://github.com/godotengine/godot/issues/94517

If you have any idea how to fix it, let me know!

**Comment by graugraugrau** [comment]
I had a quick look, cause I was not able to reproduce the problem on my Linux/Gnome machine. I tried it on Windows and can confirm the issue, that is described here. Just want to point out that it's also OS specific.

In addition I also noticed another thing that might be only slightly related. On Windows a double tap of Circumflex closes the console. On Linux the console is only closable with Escape, as soon as the console is visible. So there might also be a mismatch here.

**Comment by winkensjw** [comment]
>  How is this even supposed to behave under normal circumstances? This is so weird.

Normally you'd type it before typing a vowel and it does this: ê. These are called dead keys: https://en.wikipedia.org/wiki/Dead_key

Well, there is no direct list of dead keys in this wikipedia article (just in text) and asking AI for Keys in Godot that are dead keys yields this which seems about right skimming the wikipedia article. Its probably a bit off, and for sure not all of these identifiers AI hallucinated are available in godot:

1. KEY_QUOTELEFT (`)
2. KEY_ACUTE (´)
3. KEY_CIRCUMFLEX (^)
4. KEY_DIAERESIS (¨)
5. KEY_TILDE (~)
6. KEY_GRAVE (` alternative name)
7. KEY_APOSTROPHE (') (acts as a dead key in some layouts)
8. KEY_CEDILLA (,) (used in some languages)
9. KEY_MACRON (¯) (in extended layouts)
10. KEY_BREVE (˘) (less common)

Imho it's pretty hard to work around this because i would guess this is somewhat deep in the OS, which is why it may not happen on linux. One Idea may be to use another key (combination) altogether. But as the author you probably got that key down in muscle memory, and it in fact is very easy to reach for your keyboard locale, which I respect, so I suggest to make the key an overridable option via an own variable and note this down in the README.md. I can do a PR for you if you want me to, since I am german myself and run into this issue.
